src/Element/EntityAutocompleteMultiple.php

```diff
@@ -0,0 +1,238 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\oe_corporate_site_info\Element;
+
+use Drupal\Component\Utility\Html;
+use Drupal\Component\Utility\NestedArray;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\Core\Render\Element\FormElement;
+
+/**
+ * Provides an multiple entity autocomplete form element.
+ *
+ * The autocomplete form element allows users to select one or multiple
+ * entities, which can come from all or specific bundles of an entity type.
+ * Some features and behavior of this form element inherited from
+ * the Autocomplete widget.
+ *
+ * @see \Drupal\Core\Field\Plugin\Field\FieldWidget\EntityReferenceAutocompleteWidget
+ *
+ * Properties:
+ * - #add_more_title: (optional) Title of the "add more" button.
+ *
+ * Usage example:
+ * @code
+ * $form['my_element'] = [
+ *  '#type' => 'entity_autocomplete_multiple',
+ *  '#target_type' => 'node',
+ *  '#tags' => TRUE,
+ *  '#default_value' => [
+ *    [
+ *      'target' => $entity,
+ *      'weight' => $weight_int,
+ *    ],
+ *  ],
+ *  '#selection_handler' => 'default',
+ *  '#selection_settings' => [
+ *    'target_bundles' => ['article', 'page'],
+ *   ],
+ *  '#autocreate' => [
+ *    'bundle' => 'article',
+ *    'uid' => <a valid user ID>,
+ *   ],
+ * ];
+ * @endcode
+ *
+ * @see \Drupal\Core\Entity\Plugin\EntityReferenceSelection\DefaultSelection
+ * @see \Drupal\Core\Entity\Element\EntityAutocomplete
+ *
+ * @FormElement("entity_autocomplete_multiple")
```

Comment:
```suggestion
 * @FormElement("oe_corporate_site_info_entity_autocomplete_multiple")
```
Let's prefix the type with the module's name for now.

Reply:
done